src/launcher/plugin.ts

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import {
+  Widget
+} from 'phosphor/lib/ui/widget';
+
+import {
+  h, VNode
+} from 'phosphor/lib/ui/vdom';
+
+import {
+  VDomModel, VDomWidget
+} from '../common/vdom';
+
+import {
+  ObservableList
+} from '../common/observablelist';
+
+import {
+  DisposableDelegate, IDisposable
+} from 'phosphor/lib/core/disposable';
+
+import {
+  JupyterLab, JupyterLabPlugin
+} from '../application';
+
+import {
+  ICommandPalette
+} from '../commandpalette';
+
+import {
+  IPathTracker
+} from '../filebrowser';
+
+import {
+  IServiceManager
+} from '../services';
+
+/**
+ * The landing page extension.
+ */
+export
+const launcherExtension: JupyterLabPlugin<void> = {
+  id: 'jupyter.extensions.launcher',
+  requires: [IServiceManager, IPathTracker, ICommandPalette],
+  activate: activateLauncher,
+  autoStart: true
+};
+
+
+
+function activateLauncher(app: JupyterLab, services: IServiceManager, pathTracker: IPathTracker, palette: ICommandPalette): void {
+  let launcherModel = new LauncherModel(pathTracker);
+  let launcherWidget = new LauncherWidget()
+
+  launcherWidget.model = launcherModel;
+  launcherWidget.id = 'landing-jupyterlab-widget';
+  launcherWidget.title.label = 'Launcher';
+  launcherWidget.addClass('jp-Launcher');
+
+  // hardcoded defaults
+  let names = [
+      'Notebook',
+      'Console',
+      'Terminal',
+      'Text Editor',
+  ];
+
+  let actions = [
+      'file-operations:new-notebook',
+      `console:create-${services.kernelspecs.default}`,
+      'terminal:create-new',
+      'file-operations:new-text-file',
+  ]
+
+  let list : IDisposable[] = []; //DEMO
```

Comment:
You mean before the demo? I think in the long run we should rip it out and provide a registration system.